packages/wrangler/src/deployment-bundle/create-worker-upload-form.ts

```diff
@@ -346,9 +348,34 @@ export function createWorkerUploadForm(worker: CfWorkerInit): FormData {
 		}
 	}
 
+	const capnp_schemas: string[] = [];
+
 	if (bindings.unsafe?.bindings) {
 		// @ts-expect-error unsafe bindings don't need to match a specific type here
 		metadataBindings.push(...bindings.unsafe.bindings);
+		for (const binding of bindings.unsafe.bindings) {
+			if (binding.capnp_schema) {
+				capnp_schemas.push(binding.capnp_schema);
+			}
+		}
+	}
+
+	if (bindings.logfwdr?.capnp_schema) {
+		capnp_schemas.push(bindings.logfwdr.capnp_schema);
+	}
+
+	let capnpSchemaOutputFile: string | undefined;
+
+	if (capnp_schemas.length) {
+		const capnpOutput = callCapnp(capnp_schemas, capnp_src_prefix);
+		capnpSchemaOutputFile = `./capnp-${Date.now()}.compiled`;
+
+		formData.set(
+			capnpSchemaOutputFile,
+			new File([capnpOutput], capnpSchemaOutputFile, {
+				type: "application/octet-stream",
+			})
+		);
```

Comment:
This function (`createWorkerUploadForm`) is pretty long already—it would be nice to see the capnp stuff extracted out into a separate helper function